src/pathpyG/core/TemporalGraph.py

```diff
@@ -4,92 +4,71 @@
 import torch
 import torch_geometric
 import torch_geometric.utils
-from torch_geometric.data import TemporalData
+from torch_geometric.data import TemporalData, Data
 from torch import IntTensor
 
 import datetime
 from time import mktime
 
 from pathpyG import Graph
+from pathpyG.core.IndexMap import IndexMap
 from pathpyG.utils.config import config
 
 
 class TemporalGraph(Graph):
-    def __init__(self, edge_index, t, node_id=[], **kwargs):
-        """Creates an instance of a temporal graph with given edge index and timestamps of edges"""
-
-        assert len(node_id) == len(set(node_id)), "node_id entries must be unique"
-
-        # sort edges by timestamp and reorder edge_index accordingly
-        t_sorted, indices = torch.sort(torch.tensor(t).to(config["torch"]["device"]))
-
-        if len(node_id) == 0:
-            self.data = TemporalData(
-                src=edge_index[0][indices],
-                dst=edge_index[1][indices],
-                t=t_sorted,
-                node_id=[],
-                **kwargs,
-            )
-        else:
-            self.data = TemporalData(
-                src=edge_index[0][indices],
-                dst=edge_index[1][indices],
-                t=t_sorted,
-                node_id=node_id,
-                num_nodes=len(node_id),
-                **kwargs,
-            )
-
-        self.data["edge_index"] = edge_index[:, indices]
+    def __init__(self, data: TemporalData, mapping: IndexMap = None) -> None:
+        """Creates an instance of a temporal graph with given edge index and timestamps"""
+
+        # sort edges by timestamp
+        # Note: function sort_by_time menmtioned in pyG documentation does not exist
```

Comment:
The default pyG documentation points to `latest`, i.e. the current version of the `main` branch in GitHub. The get the documentation for the _latest release_, you need to choose a specific version or `stable` using the version selector at the bottom left of the documentation page. Thus, since `sort_by_time` is only documented in `latest`, it will probably be added in the next release.

Reply:
alright, we'll update the code later then to use the `TemporalData` function